### ISSUE TITLE
Theoretical limit on number of bi directional connections

### ISSUE THREAD

**Issue by jjalan** [bigcommerce/gruf#133]
I am unable to find this information easily but I would appreciate pointers on this.

We have a system where there is a 1 GRPC server and several GRPC client connecting to that server. Each client would keep a bi-directional stream open with server once connected for ever (until the client or server process is killed). Each client frequently every second or two sends a message to server and server does many database update (using ActiveRecords).

My question is:

1. How many such connections can a gruf server handle in parallel or on what factors does this number depends on?

I am also seeing another issue (not related to gruf) but probably connections with ActiveRecord is sometime the operation over GRPC connection fails because of `could not obtain a connection from the pool within 5.000 seconds (waited 5.083 seconds); all pooled connections were in use`. I know this is due to the fact that number of SQL connection ActiveRecord is able to make.

**Comment by billthompson** [comment]
Thanks for this question @jjalan. I added some tips to the [GRPC::RpcServer configuration](https://github.com/bigcommerce/gruf/wiki/Server:-Installation#grpcrpcserver-configuration) wiki. 

Can you please take a look and let me know if this answers your questions?

**Comment by jjalan** [comment]
@billthompson - That definitely helps. Thank you.

To ensure I have this right, if I have 100 GRPC client that keep bidirectional stream open with gruf server a time, these would be 100 concurrent requests and hence the thread pool size should be configured to 100.

And if within each thread on gruf server, say it spuns off 2 more threads (to do two different work in parallel), Activerecord connection pool size should be set to 200?

Does this sound right?

**Comment by billthompson** [comment]
Yep! You've got it @jjalan  :) 

Please let me know if you have any other questions.
